pkg/cmd/auth/shared/login_flow.go

```diff
@@ -89,11 +92,23 @@ func Login(opts *LoginOptions) error {
 			if keyChoice < len(pubKeys) {
 				keyToUpload = pubKeys[keyChoice]
 			}
-		} else {
-			var err error
-			keyToUpload, err = opts.sshContext.generateSSHKey()
+		} else if opts.sshContext.HasKeygen() {
+			var sshChoice bool
+			err := prompt.SurveyAskOne(&survey.Confirm{
+				Message: "Generate a new SSH key to add to your GitHub account?",
+				Default: true,
+			}, &sshChoice)
+
 			if err != nil {
-				return err
+				return fmt.Errorf("could not prompt: %w", err)
+			}
+
+			if sshChoice {
+				keyPair, err := opts.sshContext.GenerateSSHKey("id_ed25519", ssh.WithPassphrasePrompt(promptForSshKeyPassphrase))
```

Comment:
I don't that it's worth building an API to optionally inject an interactive passphrase prompt into GenerateSSHKey. Instead, GenerateSSHKey could just be invoked with a passphrase `string` argument that represents the passphrase that was already prompted for. The caller of GenerateSSHKey should be responsible for prompting the user beforehand.

Reply:
The reason I did it this way was to keep the semantics of `gh auth login` the same - there are some cases where `GenerateSSHKey` will bail out before if reaches the prompt (keygen doesn't exist for example) and so the user experience there is a bit flaky - you get prompted for the passphrase and then silently the app doesn't actually create anything because keygen wasn't found. 

That can be worked around by pushing the `HasKeygen` check out to callers, but I wasn't a huge fan of that as it bloated the code needed to call this function.

Reply:
Looked a little more closely and I this isn't too bad to push some things around and not have the prompt inside the function, will update